scripts/west_commands/runners/nrfjprog.py

```diff
@@ -8,10 +8,30 @@
 import os
 import shlex
 import sys
-from re import fullmatch, escape
+from re import fullmatch, match, escape
 
 from runners.core import ZephyrBinaryRunner, RunnerCaps
 
+# Helper function for inspecting hex files.
+# has_region returns True if hex file has any contents in a specific region
+# region_filter is a callable that takes an address as argument and
+# returns True if that address is in the region in question
+def has_region(region_filter, hex_file):
```

Comment:
Wouldn't it be better to use intelhex and inspect the `segments` ?

Reply:
I opted not to add intelhex as a dependency.

Reply:
Its already pulled in in `requirements-base.txt`

Reply:
It's not installed when running the tests in CI. I'd say this works for now, if we need more parsing later we can fix the CI then.

Reply:
> It's not installed when running the tests in CI

Adding this ought to fix that:

```
diff --git a/.github/workflows/west_cmds.yml b/.github/workflows/west_cmds.yml
index c7abb243a4..99c55569f9 100644
--- a/.github/workflows/west_cmds.yml
+++ b/.github/workflows/west_cmds.yml
@@ -55,7 +55,7 @@ jobs:
     - name: install pytest
       run: |
         pip3 install wheel
-        pip3 install pytest west pyelftools canopen progress mypy
+        pip3 install pytest west pyelftools canopen progress mypy intelhex
     - name: run pytest-win
       if: runner.os == 'Windows'
       run: |
```

Reply:
Thanks, done now.

Reply:
Thanks, done now.